File: ptm_task_v2/src/infrastructure/store/model/note_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

import "gorm.io/datatypes"

type NoteModel struct {
	BaseModel
	UserID   int64 `gorm:"not null;index:idx_note_user"`
	TenantID int64 `gorm:"not null"`

	TaskID    *int64 `gorm:"index:idx_note_task"`
	ProjectID *int64 `gorm:"index:idx_note_project"`

	Content      string  `gorm:"type:text;not null"`
	ContentPlain *string `gorm:"type:text"`

	Attachments datatypes.JSON `gorm:"type:jsonb"`

	IsPinned bool `gorm:"not null;default:false"`

	ActiveStatus string `gorm:"type:varchar(20);not null;default:'ACTIVE'"`
}

func (NoteModel) TableName() string {
	return "notes"
}

File: crm/src/main/java/serp/project/crm/infrastructure/store/model/NoteModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.crm.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "notes")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class NoteModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "entity_type", nullable = false, length = 50)
    private String entityType;

    @Column(name = "entity_id", nullable = false)
    private Long entityId;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;
}
